File: cgPr/cgv5fin/CGProject/Interfaced/clsEntry.vb
Imports System.Text
Public Class clsEntry
    Inherits BasicFrame
    Implements Frame

    Public Overrides Sub GenCreate_Beforelines()
        outText.AppendLine("TableName = " & q & tablename & q & ";")
    End Sub
    Public Overrides Sub GenCreate_loop_Fistline()
        outText.AppendFormat("ColumnList.AddNewEditColumn(" & q & "{0}" & q & ",{1}, {2} ,true ,false );", dbname, CrudeType, controlName)
        outText.AppendLine()
    End Sub
    Public Overrides Sub GenCreate_loop()
        outText.AppendFormat("ColumnList.AddNewEditColumn(" & q & "{0}" & q & ",{1}, {2} );", dbname, CrudeType, controlName)
        outText.AppendLine()
        If Cntrol = "lblComboBox" Then
            outText.AppendFormat("ComboBoxDataLoader.LoadData(" & q & "Id" & q & ", " & q & "Title" & q & ", " & q & "{1}" & q & ", {0}.cmbBox , true);", controlName, constraint)
        End If
        If Cntrol = "ComboBox" Then
            outText.AppendFormat("ComboBoxDataLoader.LoadData(" & q & "Id" & q & ", " & q & "Title" & q & ", " & q & "{1}" & q & ", {0}, true);", controlName, constraint)
        End If
        outText.AppendLine()
      
    End Sub

    Public Overrides Sub GenCreate_loop_Lastline()
        GenCreate_loop()
    End Sub
    Public Overrides Sub GenCreate_Afterlines()
       
    End Sub
End Class

File: Tools/DevCg/myCg/clsEntry.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using ICg;

namespace myCg
{
   
    public class clsEntry : BasicFrame, Frame
    {

        public override void GenCreate_Beforelines()
        {
            outText.AppendLine("TableName = \"" +  tablename  +"\";" );
        }
        public override void GenCreate_loop_Fistline()
        {
            if (type.ToLower() == "nodb")
            {
                return;
            }
            outText.AppendFormat("ColumnList.AddNewEditColumn(\"{0}\",{1}, {2} ,true ,false );", dbName, crudeType, controlName);
            outText.AppendLine();
        }
        public override void GenCreate_loop()
        {
            if (type.ToLower() == "nodb")
            {
                return;
            }
            outText.AppendFormat("ColumnList.AddNewEditColumn(\"{0}\",{1}, {2} );", dbName, crudeType, controlName);
            if (cntrol == "lblComboBox")
            {
                outText.AppendLine();
                outText.AppendFormat("ComboBoxDataLoader.LoadData(\"Id\", \"Title\", \"{1}\", {0}.cmbBox , true);", controlName, constraint);
            }
            if (cntrol == "ComboBox")
            {
                outText.AppendLine();
                outText.AppendFormat("ComboBoxDataLoader.LoadData(\"Id\",\"Title\",\"{1}\", {0}, true);", controlName, constraint);
            }
            outText.AppendLine();

        }

        public override void GenCreate_loop_Lastline()
        {
            GenCreate_loop();
        }
        public override void GenCreate_Afterlines()
        {

        }
    }
}
